area_detector_handlers/tests/test_hdf5.py

```diff
@@ -11,3 +13,13 @@ def test_hdf5(hdf5_files, handler):
             d = h(point_number=frame)
             assert d.shape == expected_shape
             assert np.all(d == frame)
+
+
+@select_handler("AD_HDF5")
+def test_hdf5_slicing_bound_exception(hdf5_files, handler):
+    (rpath, kwargs), (N_rows, N_cols, N_points, fpp) = hdf5_files
+    N_points = 10
```

Comment:
Redefining local variables like this can make debugging difficult because now there is no way to access the _correct_ value of `N_points`, defined on the line above this one. Can you define a new variable? It might also be useful to `assert` that the new (wrong) number of points is greater than `N_points`, just in case the test dataset changes in the future.